pylint/checkers/variables.py

```diff
@@ -2036,7 +2036,7 @@ def _in_lambda_or_comprehension_body(
             parent = parent.parent
         return False
 
-    # pylint: disable = too-many-statements, too-many-branches
```

Comment:
Nice! As the warnings indicate, `_is_variable_violation` is one of the nastiest functions in the codebase, and this is a measurable improvement.